src/System.Private.CoreLib/shared/System/Exception.cs

```diff
@@ -155,10 +155,12 @@ public override string ToString()
                 "   " + SR.Exception_EndOfInnerExceptionStack;
             }
 
+            s += Environment.NewLine;
+
             string? stackTrace = StackTrace;
             if (stackTrace != null)
             {
-                s += Environment.NewLine + stackTrace;
+                s += stackTrace;
```

Comment:
Why is this change an improvement?

Before this change, `ToString` always returned a string that did not end with a new line. After this change, it may or may not end with a new line depending on whether the exception got a stacktrace.

Reply:
Hm, I'll tweak that.

Reply:
OK, and added cases for exception with no stack with and without inner exception.

The output of [this ](https://gist.github.com/danmosemsft/2d2add4b597c645c954772920b31e05c) program is now [this](https://gist.github.com/danmosemsft/f2bdd4480affea2358efe6c6ef5fd0be).